### PR TITLE
Android: Prevent potential NPEs and improve nullability handling

### PR DESCRIPTION
**Pull request by melquiadess**
## Context

This PR aims to improve nullability handling in Godot's Android `lib` module:

- Kotlin's `!!` operator is used in a few lines, whereas elsewhere the safer `?.` is used - this is now replaced with the safe `?.` (or `?.let`) operator consistently; (usually `!!` would only be used very rarely, and with a good reason - there is one place left in this PR where I believe `!!` makes sense -> `GodotActivity.kt, L79`)
- Java's `if (x != null)` replaced with idiomatic Kotlin's `?.`
- `FileAccessHandler.renameFile()` accepted nullable args, (and so its use sites were again having `!!` for the args passed to this function),  but it can be inferred that the args the function receives are non-nullable, therefore we can get rid of nullability in those args (and so the `!!` use) 
- `Godot.getRotatedValues()` - this could produce an NPE in `onSensorChanged()` in some situations - fixed

## There are 2 outcomes from the above:
- safer code (potential NPEs eliminated, non-nullable function args)
- more consistent code (Kotlin conventions followed more closely)

### CODE REVIEW

File: platform/android/java/lib/src/org/godotengine/godot/GodotActivity.kt [hunk 1]
```diff
@@ -93,22 +94,26 @@ abstract class GodotActivity : FragmentActivity(), GodotHost {
 	}
 
 	private fun terminateGodotInstance(instance: Godot) {
-		if (godotFragment != null && instance === godotFragment!!.godot) {
-			Log.v(TAG, "Force quitting Godot instance")
-			ProcessPhoenix.forceQuit(this)
+		godotFragment?.let {
+			if (instance == it.godot) {
```
Comment (m4gr3d) [inline platform/android/java/lib/src/org/godotengine/godot/GodotActivity.kt:98]: Here the original logic has a triple equal `===` for referential equality; looks like it was inadvertently converted to a double equals `==`.

File: platform/android/java/lib/src/org/godotengine/godot/GodotActivity.kt [hunk 2]
```diff
@@ -93,22 +94,26 @@ abstract class GodotActivity : FragmentActivity(), GodotHost {
...
-			if (godotFragment != null && instance === godotFragment!!.godot) {
-				// It's very hard to properly de-initialize Godot on Android to restart the game
-				// from scratch. Therefore, we need to kill the whole app process and relaunch it.
-				//
-				// Restarting only the activity, wouldn't be enough unless it did proper cleanup (including
-				// releasing and reloading native libs or resetting their state somehow and clearing static data).
-				Log.v(TAG, "Restarting Godot instance...")
-				ProcessPhoenix.triggerRebirth(this)
+			godotFragment?.let {
+				if (instance == it.godot) {
```
Comment (m4gr3d) [inline platform/android/java/lib/src/org/godotengine/godot/GodotActivity.kt:108]: Similar comment as above; the original logic has a triple equal `===` for referential equality; looks like it was inadvertently converted to a double equals `==`.

**Review by m4gr3d — CHANGES_REQUESTED** [review]
The cleanup looks good!

Added some comments to address and it should be good to go.

File: platform/android/java/lib/src/org/godotengine/godot/GodotActivity.kt (hunk 1 above)
Comment (melquiadess) [reply platform/android/java/lib/src/org/godotengine/godot/GodotActivity.kt:98]: Ah, good catch, I missed that, thanks! Will fix...

Comment (melquiadess) [reply platform/android/java/lib/src/org/godotengine/godot/GodotActivity.kt]: Fixed

File: platform/android/java/lib/src/org/godotengine/godot/GodotActivity.kt (hunk 2 above)
Comment (melquiadess) [reply platform/android/java/lib/src/org/godotengine/godot/GodotActivity.kt]: Fixed, thanks

**Review by m4gr3d — APPROVED** [review]
Looks good!

**Comment by melquiadess** [comment]
> Looks good!

Thanks for a quick review

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**